src/grblogtools/parsers/nodelog.py

```diff
@@ -93,14 +87,38 @@ def parse(self, line: str) -> bool:
             if match:
                 self._progress.append(typeconvert_groupdict(match))
                 return True
+
         match = self.tree_search_explored.match(line)
         if match:
-            self._summary.update(typeconvert_groupdict(match))
+            entry = typeconvert_groupdict(match)
+            self._summary.update(entry)
+            self._progress_last_entry.update(
+                {
+                    "CurrentNode": entry["NodeCount"],
+                    "Time": entry["Runtime"],
+                }
+            )
             return True
+
+        match = self.tree_search_termination.match(line)
+        if match:
+            entry = typeconvert_groupdict(match)
+            self._summary.update(entry)
+            self._progress_last_entry.update(
```

Comment:
I think it's simpler to populate this line in get_progress() rather than tracking another variable attribute.

Reply:
Yes, I saw your changes. I thought this way is more clear where each pieces is added, but your version is shorter and cleaner.